tests/CMakeLists.txt

```diff
@@ -49,6 +49,9 @@ target_compile_options(test_main PUBLIC
     # https://github.com/nlohmann/json/issues/1114
     $<$<CXX_COMPILER_ID:MSVC>:/bigobj> $<$<BOOL:${MINGW}>:-Wa,-mbig-obj>
 
+    # https://github.com/nlohmann/json/pull/3229
+    $<$<CXX_COMPILER_ID:Intel>:-diag-disable=2196>
```

Comment:
This may need a version check. It's `-wd2196` for older versions but I'm not familiar with ICPC, so I don't know since when the new flag is available. :man_shrugging:

Reply:
Also 1786 to suppress deprecation warnings. We're not enabling warnings as errors (should we?), so it'd be just for cleaner output.

Reply:
I couldn't find a version number in the documentation.